staking/programs/staking/src/state/vesting.rs

```diff
@@ -235,6 +235,65 @@ impl VestingSchedule {
             amount,
         }))
     }
+
+    pub fn split_vesting_schedule(
+        &self,
+        remaining_amount: u64,
+        transferred_amount: u64,
+        total_amount: u64,
+    ) -> Result<(VestingSchedule, VestingSchedule)> {
+        require!(
+            transferred_amount
+                .checked_add(remaining_amount)
+                .ok_or(ErrorCode::Other)?
+                == total_amount,
+            ErrorCode::SanityCheckFailed
+        );
+        match self {
+            VestingSchedule::FullyVested => {
+                Ok((VestingSchedule::FullyVested, VestingSchedule::FullyVested))
+            }
+            VestingSchedule::PeriodicVesting {
+                initial_balance,
+                start_date,
+                period_duration,
+                num_periods,
+            } => Ok((
+                VestingSchedule::PeriodicVesting {
+                    initial_balance: ((remaining_amount as u128) * (*initial_balance as u128)
```

Comment:
is it ok that this arithmetic loses precision? (document why?)

Reply:
actually i think this is a bug. it seems like the two initial_balance fields should sum up to the original initial_balance, which doesn't currently happen.